### PR TITLE
Nested Query support

### PR DESCRIPTION
**Pull request by ArnaudT**
add  `NestedQuery` to query nested objects / docs
https://www.elastic.co/guide/en/elasticsearch/reference/2.3/query-dsl-nested-query.html

PR for #56

### CODE REVIEW

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala [hunk 1]
```diff
@@ -181,6 +181,11 @@ trait MappingDsl extends DslCommons {
   }
 
   case class CompletionContext(path: String)
+
+  case object NestedFieldsMapping extends FieldMapping {
```
Comment (rcoh) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala]: `NestedFieldMapping` to match other non-plural mappings

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala [hunk 2]
```diff
@@ -484,8 +484,25 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
...
     }
+
+    "Support NestedQuery" in {
+      // https://www.elastic.co/guide/en/elasticsearch/reference/2.3/query-dsl-nested-query.html
+      val metadataMapping = Mapping(tpe, IndexMapping(Map("user" -> NestedFieldsMapping), EnabledFieldMapping(true), None))
+      val mappingFuture = restClient.putMapping(index, tpe, metadataMapping)
+      whenReady(mappingFuture) { _ => refresh() }
+      val userDoc = List(Map("first" -> "john", "last" -> "Smith"), Map("first" -> "Alice", "last" -> "White"))
+      val matchDoc = Document("matchDoc", Map("user" -> userDoc))
+      val matchNotInsertionFuture = restClient.index(index, tpe, matchDoc)
```
Comment (rcoh) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: `matchDocInsertionFuture`?

**Comment by rcoh** [comment]
A couple rename requests, but looks good otherwise. Thanks for contributing!

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala (hunk 1 above)
Comment (ArnaudT) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/MappingDsl.scala]: yes true. I will rename it.

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala (hunk 2 above)
Comment (ArnaudT) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: yes. I'll rename it.

**Comment by rcoh** [comment]
LGTM thanks!

**Pull request merged**